storage/client_raft_test.go

```diff
@@ -269,3 +270,79 @@ func TestFailedReplicaChange(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// We can truncate the old log entries and a new replica will be brought up from a snapshot.
+func TestReplicateAfterTruncation(t *testing.T) {
+	mtc := multiTestContext{}
+	mtc.Start(t, 2)
+	defer mtc.Stop()
+
+	rng, err := mtc.stores[0].GetRange(1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Issue a command on the first node before replicating.
+	incArgs, incResp := incrementArgs([]byte("a"), 5, 1, mtc.stores[0].StoreID())
+	if err := mtc.stores[0].ExecuteCmd(proto.Increment, incArgs, incResp); err != nil {
+		t.Fatal(err)
+	}
+
+	// Get that command's log index.
+	index, err := rng.LastIndex()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Truncate the log.
+	truncArgs, truncResp := internalTruncateLogArgs(index, 1, mtc.stores[0].StoreID())
```

Comment:
this truncates everything `< index`, right? Do we want to put index+1 here to truncate the increment as well?